scripts/train_loops/mbrl_loop.py

```diff
@@ -0,0 +1,211 @@
+import logging
+import time
+from cares_reinforcement_learning.util import helpers as hlp
+from cares_reinforcement_learning.util.configurations import (
+    AlgorithmConfig,
+    TrainingConfig,
+)
+
+
+def evaluate_policy_network(
+    env, agent, config: TrainingConfig, record=None, total_steps=0
+):
+    """
+    This function evaluate the agent and world model at a fixed interval
+    (10000 for now).
+    Cumulative rewards are averaged across 10 episodes.
+
+    Mean Square Error for world model is used to evaluate a world model.
+
+    """
+    if record is not None:
+        frame = env.grab_frame()
+        record.start_video(total_steps + 1, frame)
+    number_eval_episodes = int(config.number_eval_episodes)
+    state = env.reset()
+    for eval_episode_counter in range(number_eval_episodes):
+        episode_timesteps = 0
+        episode_reward = 0
+        episode_num = 0
+        done = False
+        truncated = False
+        while not done and not truncated:
+            episode_timesteps += 1
+            action = agent.select_action_from_policy(state, evaluation=True)
+            action_env = hlp.denormalize(
+                action, env.max_action_value, env.min_action_value
+            )
+            next_state, reward, done, truncated = env.step(action_env)
+            state = next_state
+            episode_reward += reward
+            if eval_episode_counter == 0 and record is not None:
+                frame = env.grab_frame()
+                record.log_video(frame)
+            if done or truncated:
+                if record is not None:
+                    record.log_eval(
+                        total_steps=total_steps + 1,
+                        episode=eval_episode_counter + 1,
+                        episode_reward=episode_reward,
+                        display=True,
+                    )
+                # Reset environment
+                state = env.reset()
+                episode_reward = 0
+                episode_timesteps = 0
+                episode_num += 1
+    record.stop_video()
+
+
+def policy_based_train(
```

Comment:
Nope, there should be a separate file for discrete (value) based MBRL in the future. Also, I changed based on this policy_training loop to show more clearly where it will be changed.